{"cells":[{"cell_type":"markdown","id":"56248b52-5e0e-4dc9-81e8-8121b484bf3a","metadata":{"id":"56248b52-5e0e-4dc9-81e8-8121b484bf3a"},"source":["# **Generating Protein Structure Networks from PDB files Using the RING server**\n","---"]},{"cell_type":"markdown","id":"02c909a1-4f8b-434f-be57-cdf3ce669ff6","metadata":{"id":"02c909a1-4f8b-434f-be57-cdf3ce669ff6"},"source":["![1yok_workflow.png](attachment:3df0ad85-c1c9-421e-9ea7-c5923d525162.png)"]},{"cell_type":"markdown","id":"fd659677-91a0-4866-9370-44d78133f743","metadata":{"id":"fd659677-91a0-4866-9370-44d78133f743"},"source":["<font size='4'>The method employed in this project for generating protein structure networks begins with a folder or list of relevant Protein Data Bank (PDB) structure files. A PDB structure file is a standard file format used to represent three-dimensional molecular structures, primarily of proteins and nucleic acids. It contains detailed information about the atomic coordinates, chemical components, and molecular interactions within the structure. PDB structure files are crucial for understanding the molecular architecture and interactions of biological molecules and are widely used in structural biology, bioinformatics, and drug discovery research. They adhere to a standardized format maintained by the [Worldwide Protein Data Bank](https://www.rcsb.org/) to ensure compatibility and interoperability across various computational tools and databases. Here's what a typical PDB structure file contains:</font>"]},{"cell_type":"code","execution_count":null,"id":"d2580043-eebe-48d8-a39f-2e39faa1a6a5","metadata":{"id":"d2580043-eebe-48d8-a39f-2e39faa1a6a5","outputId":"3f26ea5c-6969-4528-e706-9a5653601158"},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>Model</th>\n","      <th>Chain</th>\n","      <th>Residue_Number</th>\n","      <th>Residue_Name</th>\n","      <th>Atom_Name</th>\n","      <th>X</th>\n","      <th>Y</th>\n","      <th>Z</th>\n","      <th>Occupancy</th>\n","      <th>B_Factor</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>0</td>\n","      <td>A</td>\n","      <td>300</td>\n","      <td>SER</td>\n","      <td>N</td>\n","      <td>28.985001</td>\n","      <td>54.266998</td>\n","      <td>34.372002</td>\n","      <td>1.0</td>\n","      <td>88.63</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>0</td>\n","      <td>A</td>\n","      <td>300</td>\n","      <td>SER</td>\n","      <td>CA</td>\n","      <td>27.739000</td>\n","      <td>54.654999</td>\n","      <td>33.648998</td>\n","      <td>1.0</td>\n","      <td>88.62</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>0</td>\n","      <td>A</td>\n","      <td>300</td>\n","      <td>SER</td>\n","      <td>C</td>\n","      <td>27.105000</td>\n","      <td>53.480999</td>\n","      <td>32.896999</td>\n","      <td>1.0</td>\n","      <td>89.53</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>0</td>\n","      <td>A</td>\n","      <td>300</td>\n","      <td>SER</td>\n","      <td>O</td>\n","      <td>27.492001</td>\n","      <td>53.158001</td>\n","      <td>31.774000</td>\n","      <td>1.0</td>\n","      <td>90.12</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>0</td>\n","      <td>A</td>\n","      <td>300</td>\n","      <td>SER</td>\n","      <td>CB</td>\n","      <td>28.028999</td>\n","      <td>55.799000</td>\n","      <td>32.671001</td>\n","      <td>1.0</td>\n","      <td>87.83</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>2589</th>\n","      <td>0</td>\n","      <td>N</td>\n","      <td>1</td>\n","      <td>UNK</td>\n","      <td>I</td>\n","      <td>26.409000</td>\n","      <td>23.811001</td>\n","      <td>36.709000</td>\n","      <td>0.0</td>\n","      <td>0.00</td>\n","    </tr>\n","    <tr>\n","      <th>2590</th>\n","      <td>0</td>\n","      <td>N</td>\n","      <td>1</td>\n","      <td>UNK</td>\n","      <td>S</td>\n","      <td>30.770000</td>\n","      <td>21.639999</td>\n","      <td>30.334999</td>\n","      <td>0.0</td>\n","      <td>0.00</td>\n","    </tr>\n","    <tr>\n","      <th>2591</th>\n","      <td>0</td>\n","      <td>N</td>\n","      <td>1</td>\n","      <td>UNK</td>\n","      <td>H</td>\n","      <td>23.517000</td>\n","      <td>23.597000</td>\n","      <td>28.778000</td>\n","      <td>0.0</td>\n","      <td>0.00</td>\n","    </tr>\n","    <tr>\n","      <th>2592</th>\n","      <td>0</td>\n","      <td>N</td>\n","      <td>1</td>\n","      <td>UNK</td>\n","      <td>HN</td>\n","      <td>34.470001</td>\n","      <td>26.077999</td>\n","      <td>34.348999</td>\n","      <td>1.0</td>\n","      <td>0.00</td>\n","    </tr>\n","    <tr>\n","      <th>2593</th>\n","      <td>0</td>\n","      <td>N</td>\n","      <td>1</td>\n","      <td>UNK</td>\n","      <td>HO</td>\n","      <td>24.819000</td>\n","      <td>25.201000</td>\n","      <td>36.011002</td>\n","      <td>1.0</td>\n","      <td>0.00</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>2594 rows Ã— 10 columns</p>\n","</div>"],"text/plain":["      Model Chain  Residue_Number Residue_Name Atom_Name          X  \\\n","0         0     A             300          SER         N  28.985001   \n","1         0     A             300          SER        CA  27.739000   \n","2         0     A             300          SER         C  27.105000   \n","3         0     A             300          SER         O  27.492001   \n","4         0     A             300          SER        CB  28.028999   \n","...     ...   ...             ...          ...       ...        ...   \n","2589      0     N               1          UNK         I  26.409000   \n","2590      0     N               1          UNK         S  30.770000   \n","2591      0     N               1          UNK         H  23.517000   \n","2592      0     N               1          UNK        HN  34.470001   \n","2593      0     N               1          UNK        HO  24.819000   \n","\n","              Y          Z  Occupancy  B_Factor  \n","0     54.266998  34.372002        1.0     88.63  \n","1     54.654999  33.648998        1.0     88.62  \n","2     53.480999  32.896999        1.0     89.53  \n","3     53.158001  31.774000        1.0     90.12  \n","4     55.799000  32.671001        1.0     87.83  \n","...         ...        ...        ...       ...  \n","2589  23.811001  36.709000        0.0      0.00  \n","2590  21.639999  30.334999        0.0      0.00  \n","2591  23.597000  28.778000        0.0      0.00  \n","2592  26.077999  34.348999        1.0      0.00  \n","2593  25.201000  36.011002        1.0      0.00  \n","\n","[2594 rows x 10 columns]"]},"execution_count":5,"metadata":{},"output_type":"execute_result"}],"source":["import pandas as pd\n","from Bio.PDB import PDBParser\n","\n","def pdb_to_dataframe(pdb_file):\n","    parser = PDBParser(QUIET=True)\n","    structure = parser.get_structure('structure', pdb_file)\n","\n","    data = []\n","    for model in structure:\n","        for chain in model:\n","            for residue in chain:\n","                for atom in residue:\n","                    data.append([\n","                        model.id,\n","                        chain.id,\n","                        residue.id[1],\n","                        residue.resname,\n","                        atom.name,\n","                        atom.coord[0],\n","                        atom.coord[1],\n","                        atom.coord[2],\n","                        atom.occupancy,\n","                        atom.bfactor\n","                    ])\n","\n","    columns = [\n","        'Model', 'Chain', 'Residue_Number', 'Residue_Name',\n","        'Atom_Name', 'X', 'Y', 'Z', 'Occupancy', 'B_Factor'\n","    ]\n","\n","    df = pd.DataFrame(data, columns=columns)\n","    return df\n","\n","# Example usage:\n","pdb_file = 'structure_files/1yok.pdb'\n","pdb_df = pdb_to_dataframe(pdb_file)\n","pdb_df"]},{"cell_type":"markdown","id":"ae303f47-9f5f-421e-b1e2-3d512f63f513","metadata":{"id":"ae303f47-9f5f-421e-b1e2-3d512f63f513"},"source":["<font size='4'>The [Residue Interaction Network Generator](https://ring.biocomputingup.it/) (RING) server is a computational tool designed to analyze and visualize residue interactions in proteins. It uses protein structure data to create residue interaction networks (RINs), which represent the network of contacts and interactions between amino acid residues within a protein structure.</font>\n","\n","## **Key Features and Uses:**\n","\n","#### 1. **Network Visualization:** RING server visualizes residue interactions in a protein as a network, where nodes represent residues and edges represent interactions (e.g., hydrogen bonds, van der Waals contacts).\n","\n","#### 2. **Interaction Analysis:** It quantifies and categorizes different types of residue interactions within the protein structure.\n","\n","#### 3. **Comparative Analysis:** It allows for the comparison of residue interaction networks between different protein structures or between different conformations of the same protein.\n","\n","#### 4. **Dynamic Analysis:** It can analyze and visualize changes in residue interactions in molecular dynamics simulations or upon ligand binding.\n","\n","#### 5. **Structural Bioinformatics:** The server provides insights into the structural basis of protein function, stability, and dynamics.\n","\n","#### 6. **Predictive Modeling:** It can aid in predicting the effects of mutations or protein engineering on residue interactions and stability.\n","\n","## **Why is it Useful for Research?**\n","\n","<font size='4'>Protein structure networks are essential tools for researchers because they provide a detailed map of interactions between amino acids within a protein. This information is fundamental for understanding protein function, including enzymatic activity, molecular recognition, and signaling pathways. These networks also aid in predicting important protein properties such as stability, flexibility, and solubility, which are crucial for protein engineering and drug design. By identifying key residues and interactions, researchers can pinpoint critical sites that stabilize the protein structure or mediate specific functional roles, guiding both experimental studies and computational predictions. Comparative analysis of protein structures helps to reveal evolutionary relationships, structural motifs, and functional similarities or differences among proteins. Additionally, protein structure networks facilitate structure-based drug design by identifying potential targets for therapeutic agents. They also support molecular dynamics and simulation studies, enabling researchers to predict how mutations or environmental changes affect protein stability and function. Moreover, these networks allow for the systematic analysis of protein-protein interactions, elucidating how proteins form complexes and cooperate in biological processes. In the field of structural bioinformatics, protein structure networks serve as a foundation for developing algorithms and tools for protein structure prediction, analysis, and visualization, thereby advancing research in molecular biology and biophysics.</font>\n","\n","<font size='4'>In summary, the RING server is a powerful tool in structural bioinformatics that facilitates the analysis, visualization, and comparison of residue interaction networks within protein structures, contributing significantly to research in molecular biology, biophysics, and drug discovery.</font>"]},{"cell_type":"markdown","id":"b4a80676-e7ab-432a-bef3-bf1d8aa88612","metadata":{"id":"b4a80676-e7ab-432a-bef3-bf1d8aa88612"},"source":["### User defined functions for creating necessary folders:"]},{"cell_type":"code","execution_count":null,"id":"64f9b27e-f06f-41d0-b758-f60bf12728ba","metadata":{"id":"64f9b27e-f06f-41d0-b758-f60bf12728ba"},"outputs":[],"source":["# Function to verify or create folder\n","def ensure_folder_exists(directory_path):\n","    if not os.path.exists(directory_path):\n","        os.makedirs(directory_path)\n","        print(f\"Folder '{directory_path}' created.\")\n","    else:\n","        print(f\"Folder '{directory_path}' already exists.\")\n","\n","# Function for coverting pdb files to cif files\n","def convert_pdb_to_cif(pdb_file, cif_file):\n","    command = [\n","        \"pymol\", \"-cq\", pdb_file,\n","        \"-d\", f\"save {cif_file}, state=1; quit;\"\n","    ]\n","    try:\n","        subprocess.run(command, check=True)\n","        print(f\"Conversion successful: {pdb_file} -> {cif_file}\")\n","    except subprocess.CalledProcessError as e:\n","        print(f\"Error during conversion: {e}\")"]},{"cell_type":"markdown","id":"d6f543ba-e0ac-4985-860e-92afbf4da267","metadata":{"id":"d6f543ba-e0ac-4985-860e-92afbf4da267"},"source":["# Generate Directory of .cif Files from Directory of Selected PDB Files:\n","---"]},{"cell_type":"code","execution_count":null,"id":"37b8cd2a-744d-4bb0-af17-1b0ca444d4b6","metadata":{"id":"37b8cd2a-744d-4bb0-af17-1b0ca444d4b6"},"outputs":[],"source":["import os\n","import subprocess\n","\n","# Directories\n","input_dir = \"structure_files/\"\n","output_dir = \"cif_files/\"\n","\n","# Ensure directories exist\n","ensure_folder_exists(input_dir)\n","ensure_folder_exists(output_dir)\n","\n","# Loop over all pdb files in the input directory\n","for filename in os.listdir(input_dir):\n","    if filename.endswith(\".pdb\"):\n","        pdb_file_path = os.path.join(input_dir, filename)\n","        cif_file_name = os.path.splitext(filename)[0] + \".cif\"\n","        cif_file_path = os.path.join(output_dir, cif_file_name)\n","\n","        convert_pdb_to_cif(pdb_file_path, cif_file_path)\n"]},{"cell_type":"markdown","id":"0da0dd20-8171-46f4-959c-f99abf3e3fc9","metadata":{"id":"0da0dd20-8171-46f4-959c-f99abf3e3fc9"},"source":[]},{"cell_type":"markdown","id":"f7c2395a-7bce-41ab-9339-4f7a9b54a381","metadata":{"id":"f7c2395a-7bce-41ab-9339-4f7a9b54a381"},"source":["# RING Server API:\n","---"]},{"cell_type":"code","execution_count":null,"id":"9521c01a-3130-40dd-998b-faec91c84f71","metadata":{"id":"9521c01a-3130-40dd-998b-faec91c84f71"},"outputs":[],"source":["module_code = '''\n","\n","import requests\n","import os\n","import time\n","from threading import Thread\n","import shutil\n","\n","list_task_url = \"https://scheduler.biocomputingup.it/task/\"\n","list_script_url = \"https://scheduler.biocomputingup.it/script/\"\n","list_params_url = \"https://scheduler.biocomputingup.it/params/\"\n","\n","class Status:\n","    statusMap = {\n","        \"task has been rejected from the ws\": \"failed\",\n","        \"task has been received from the ws\": \"pending\",\n","        \"task has been created and sent to the DRM\": \"pending\",\n","        \"process status cannot be determined\": \"pending\",\n","        \"job is queued and active\": \"running\",\n","        \"job is queued and in system hold\": \"running\",\n","        \"job is queued and in user hold\": \"running\",\n","        \"job is queued and in user and system hold\": \"running\",\n","        \"job is running\": \"running\",\n","        \"job is system suspended\": \"pending\",\n","        \"job is user suspended\": \"pending\",\n","        \"job finished normally\": \"success\",\n","        \"job finished, but failed\": \"failed\",\n","        \"job has been deleted\": \"deleted\"\n","    }\n","\n","    def __init__(self, status):\n","        self.status = self.decode_status(status)\n","\n","    def __repr__(self):\n","        return self.status\n","\n","    def __eq__(self, other):\n","        return self.status == other\n","\n","    def decode_status(self, status_long):\n","        return self.statusMap[status_long]\n","\n","\n","class Task:\n","    _status: [Status, None] = None\n","    _uuid: [str, None] = None\n","\n","    def __init__(self, uuid=None, status=None):\n","        self.uuid = uuid\n","        self.status = status\n","\n","    def __repr__(self) -> str:\n","        return \"{} - {}\".format(self.uuid, self.status)\n","\n","    @property\n","    def status(self):\n","        return self._status\n","\n","    @status.setter\n","    def status(self, status):\n","        self._status = Status(status) if status is not None else status\n","\n","    @property\n","    def uuid(self):\n","        return self._uuid\n","\n","    @uuid.setter\n","    def uuid(self, uuid):\n","        self._uuid = uuid\n","\n","    def is_finished(self) -> bool:\n","        return self._status == \"failed\" or self._status == \"deleted\" or self._status == \"success\"\n","\n","\n","def check_for_job(task):\n","    try:\n","        job_url = \"{}/{}\".format(list_task_url, task.uuid)\n","\n","        while not task.is_finished():\n","            response = requests.get(job_url, timeout=5)\n","            response.raise_for_status()\n","            task.status = response.json()[\"status\"]\n","            if not task.is_finished():\n","                time.sleep(3)\n","\n","    except requests.exceptions.RequestException as err:\n","        return err\n","\n","\n","def post_job(task, file_pth, params):\n","    try:\n","        files = {'input_file': open(file_pth, 'rb')}\n","\n","        response = requests.post(list_task_url, files=files, data=params, timeout=5)\n","        response.raise_for_status()\n","        task.uuid = response.json()[\"uuid\"]\n","        task.status = response.json()[\"status\"]\n","\n","    except requests.exceptions.RequestException as err:\n","        return err\n","\n","\n","def download_results(task, extract_pth):\n","    if task.status == \"failed\":\n","        return\n","    try:\n","        output_url = \"{}/{}/{}\".format(list_task_url, task.uuid, \"download\")\n","\n","        response = requests.get(output_url, timeout=5)\n","        response.raise_for_status()\n","        file_name = response.headers[\"content-disposition\"].split(\"filename=\")[1]\n","        file_pth = \"{}/{}\".format(extract_pth, file_name)\n","\n","        with open(file_pth, \"wb\") as f:\n","            f.write(response.content)\n","\n","        shutil.unpack_archive(file_pth, extract_pth)\n","        os.remove(file_pth)\n","\n","    except requests.exceptions.RequestException as err:\n","        return err\n","\n","\n","def config_to_parameters(config):\n","    convert = {\n","        \"-g\": \"seq_sep\",\n","        \"-o\": \"len_salt\",\n","        \"-s\": \"len_ss\",\n","        \"-k\": \"len_pipi\",\n","        \"-a\": \"len_pica\",\n","        \"-b\": \"len_hbond\",\n","        \"-w\": \"len_vdw\"\n","    }\n","\n","    new_config = {}\n","\n","    for key, value in config.items():\n","        if key in convert:\n","            new_config[convert[key]] = value.strip(\"--\")\n","\n","    new_config[config[\"edges\"].strip(\"--\")] = True\n","\n","    return new_config\n","\n","\n","def run_ring_api(file_pth, run_config, tmp_dir, progress_f):\n","\n","    task = Task()\n","\n","    file_name = os.path.basename(file_pth)\n","\n","    parameters = {\n","        \"task_name\": \"ring-plugin-api\",\n","        \"original_name\": file_name\n","    }\n","\n","    parameters.update(config_to_parameters(run_config))\n","\n","    t_post_job = Thread(target=post_job, args=(task, file_pth, parameters))\n","    t_post_job.start()\n","\n","    prev_progress = 0\n","    while t_post_job.is_alive():\n","        progress_f(min([prev_progress, 15]))\n","        prev_progress += 0.01\n","\n","    t_check_job = Thread(target=check_for_job, args=(task,))\n","    t_check_job.start()\n","\n","    prev_progress = 15\n","    timer = time.time() - 5\n","    while t_check_job.is_alive():\n","        if time.time() - timer > 5:\n","            timer = time.time()\n","\n","        progress_f(min([prev_progress, 85]))\n","        prev_progress += 0.00001\n","\n","    t_download_results = Thread(target=download_results, args=(task, tmp_dir))\n","    t_download_results.start()\n","\n","    prev_progress = 85\n","\n","    while t_download_results.is_alive():\n","        progress_f(min([prev_progress, 100]))\n","        prev_progress += 0.01\n","\n","    progress_f(100)\n","'''\n","\n","with open('ring_api_module.py', 'w') as file:\n","    file.write(module_code)"]},{"cell_type":"markdown","id":"a3a7aae2-6bd7-4c6d-80f5-af0a8356ceb8","metadata":{"id":"a3a7aae2-6bd7-4c6d-80f5-af0a8356ceb8"},"source":["# Run RING API using Strict Thresholds:\n","---"]},{"cell_type":"code","execution_count":null,"id":"476af61e-ccbc-4661-b93a-86d938109142","metadata":{"id":"476af61e-ccbc-4661-b93a-86d938109142"},"outputs":[],"source":["import os\n","from ring_api_module import run_ring_api\n","\n","def progress(p):\n","    print(f\"Progress: {p:.02f}%\")\n","\n","dir_name = 'cif_files/'\n","crnt_dir = os.getcwd()\n","dir_path = os.path.join(crnt_dir, dir_name)\n","file_names = os.listdir(dir_path)\n","\n","'''\n","-g : sequence separation\n","-o : salt bridge (ionic bond)\n","-s : disulphide bond distance\n","-k : pi-pi stacking distance\n","-a : pi-cation distance\n","-b : hydrogen bond distance\n","-w : van Der Waals radii\n","'''\n","\n","run_config = {\n","    \"-g\": '3',\n","    \"-o\": '4.0',\n","    \"-s\": '2.5',\n","    \"-k\": '6.5',\n","    \"-a\": '5.0',\n","    \"-b\": '3.5',\n","    \"-w\": '0.5',\n","    \"edges\": \"--best_edge\"\n","}\n","tmp_dir = 'results/'\n","ensure_folder_exists(tmp_dir)\n","\n","cif_files = [file for file in file_names if file.endswith('.cif')]\n","\n","for file_name in cif_files:\n","    file_path = os.path.join(dir_path, file_name)\n","    run_ring_api(file_path, run_config, tmp_dir, progress)\n"]},{"cell_type":"markdown","id":"eddc65cc-8201-4738-a264-383208107e7b","metadata":{"id":"eddc65cc-8201-4738-a264-383208107e7b"},"source":["# Run RING API Using Relaxed Thresholds:\n","---"]},{"cell_type":"code","execution_count":null,"id":"743b95c3-63c2-4007-a5d2-83da9f5acb50","metadata":{"scrolled":true,"id":"743b95c3-63c2-4007-a5d2-83da9f5acb50"},"outputs":[],"source":["import os\n","from ring_api_module import run_ring_api\n","\n","def progress(p):\n","    print(f\"Progress: {p:.02f}%\")\n","\n","dir_name = 'cif_files/'\n","crnt_dir = os.getcwd()\n","dir_path = os.path.join(crnt_dir, dir_name)\n","file_names = os.listdir(dir_path)\n","\n","'''\n","-g : sequence separation\n","-o : salt bridge (ionic bond)\n","-s : disulphide bond distance\n","-k : pi-pi stacking distance\n","-a : pi-cation distance\n","-b : hydrogen bond distance\n","-w : van Der Waals radii\n","'''\n","\n","run_config = {\n","    \"-g\": '3',\n","    \"-o\": '5.0',\n","    \"-s\": '3.0',\n","    \"-k\": '7.0',\n","    \"-a\": '7.0',\n","    \"-b\": '5.5',\n","    \"-w\": '0.8',\n","    \"edges\": \"--best_edge\"\n","}\n","tmp_dir = 'results2/'\n","ensure_folder_exists(tmp_dir)\n","\n","cif_files = [file for file in file_names if file.endswith('.cif')]\n","\n","for file_name in cif_files:\n","    file_path = os.path.join(dir_path, file_name)\n","    run_ring_api(file_path, run_config, tmp_dir, progress)\n"]},{"cell_type":"markdown","id":"3a68f0f6-a952-47e5-ba2d-e043dac57747","metadata":{"id":"3a68f0f6-a952-47e5-ba2d-e043dac57747"},"source":["# Result from Executing the RING API Using Relaxed Parameters:\n","---"]},{"cell_type":"code","execution_count":null,"id":"70675672-b49b-4ae3-bce1-8b27382f6930","metadata":{"id":"70675672-b49b-4ae3-bce1-8b27382f6930","outputId":"80f1caec-5882-4653-803b-93758dad2667"},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>NodeId1</th>\n","      <th>Interaction</th>\n","      <th>NodeId2</th>\n","      <th>Distance</th>\n","      <th>Angle</th>\n","      <th>Energy</th>\n","      <th>Atom1</th>\n","      <th>Atom2</th>\n","      <th>Donor</th>\n","      <th>Positive</th>\n","      <th>Cation</th>\n","      <th>Orientation</th>\n","      <th>Model</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>A:300:_:SER</td>\n","      <td>VDW:MC_SC</td>\n","      <td>A:488:_:ASN</td>\n","      <td>3.976</td>\n","      <td>NaN</td>\n","      <td>6.0</td>\n","      <td>C</td>\n","      <td>ND2</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>A:301:_:ILE</td>\n","      <td>VDW:SC_SC</td>\n","      <td>A:305:_:ILE</td>\n","      <td>3.655</td>\n","      <td>NaN</td>\n","      <td>6.0</td>\n","      <td>CG2</td>\n","      <td>CG2</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>A:301:_:ILE</td>\n","      <td>VDW:SC_SC</td>\n","      <td>A:306:_:LEU</td>\n","      <td>3.774</td>\n","      <td>NaN</td>\n","      <td>6.0</td>\n","      <td>CB</td>\n","      <td>CD2</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>A:301:_:ILE</td>\n","      <td>VDW:SC_SC</td>\n","      <td>A:446:_:ARG</td>\n","      <td>3.918</td>\n","      <td>NaN</td>\n","      <td>6.0</td>\n","      <td>CD1</td>\n","      <td>CD</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>A:302:_:PRO</td>\n","      <td>VDW:SC_SC</td>\n","      <td>A:305:_:ILE</td>\n","      <td>4.186</td>\n","      <td>NaN</td>\n","      <td>6.0</td>\n","      <td>CB</td>\n","      <td>CD1</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>425</th>\n","      <td>C:745:_:LEU</td>\n","      <td>VDW:MC_SC</td>\n","      <td>C:748:_:LEU</td>\n","      <td>4.197</td>\n","      <td>NaN</td>\n","      <td>6.0</td>\n","      <td>C</td>\n","      <td>CB</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>426</th>\n","      <td>C:745:_:LEU</td>\n","      <td>VDW:MC_SC</td>\n","      <td>C:749:_:LEU</td>\n","      <td>3.788</td>\n","      <td>NaN</td>\n","      <td>6.0</td>\n","      <td>C</td>\n","      <td>CD2</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>427</th>\n","      <td>C:746:_:ARG</td>\n","      <td>HBOND:SC_SC</td>\n","      <td>C:750:_:ASP</td>\n","      <td>3.383</td>\n","      <td>17.538</td>\n","      <td>17.0</td>\n","      <td>NE</td>\n","      <td>OD2</td>\n","      <td>C:746:_:ARG</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>428</th>\n","      <td>C:747:_:TYR</td>\n","      <td>HBOND:SC_SC</td>\n","      <td>C:751:_:LYS</td>\n","      <td>3.541</td>\n","      <td>48.081</td>\n","      <td>17.0</td>\n","      <td>OH</td>\n","      <td>NZ</td>\n","      <td>C:747:_:TYR</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>429</th>\n","      <td>C:748:_:LEU</td>\n","      <td>VDW:MC_SC</td>\n","      <td>C:751:_:LYS</td>\n","      <td>4.188</td>\n","      <td>NaN</td>\n","      <td>6.0</td>\n","      <td>C</td>\n","      <td>CG</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>NaN</td>\n","      <td>1</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>430 rows Ã— 13 columns</p>\n","</div>"],"text/plain":["         NodeId1  Interaction      NodeId2  Distance   Angle  Energy Atom1  \\\n","0    A:300:_:SER    VDW:MC_SC  A:488:_:ASN     3.976     NaN     6.0     C   \n","1    A:301:_:ILE    VDW:SC_SC  A:305:_:ILE     3.655     NaN     6.0   CG2   \n","2    A:301:_:ILE    VDW:SC_SC  A:306:_:LEU     3.774     NaN     6.0    CB   \n","3    A:301:_:ILE    VDW:SC_SC  A:446:_:ARG     3.918     NaN     6.0   CD1   \n","4    A:302:_:PRO    VDW:SC_SC  A:305:_:ILE     4.186     NaN     6.0    CB   \n","..           ...          ...          ...       ...     ...     ...   ...   \n","425  C:745:_:LEU    VDW:MC_SC  C:748:_:LEU     4.197     NaN     6.0     C   \n","426  C:745:_:LEU    VDW:MC_SC  C:749:_:LEU     3.788     NaN     6.0     C   \n","427  C:746:_:ARG  HBOND:SC_SC  C:750:_:ASP     3.383  17.538    17.0    NE   \n","428  C:747:_:TYR  HBOND:SC_SC  C:751:_:LYS     3.541  48.081    17.0    OH   \n","429  C:748:_:LEU    VDW:MC_SC  C:751:_:LYS     4.188     NaN     6.0     C   \n","\n","    Atom2        Donor Positive Cation Orientation  Model  \n","0     ND2          NaN      NaN    NaN         NaN      1  \n","1     CG2          NaN      NaN    NaN         NaN      1  \n","2     CD2          NaN      NaN    NaN         NaN      1  \n","3      CD          NaN      NaN    NaN         NaN      1  \n","4     CD1          NaN      NaN    NaN         NaN      1  \n","..    ...          ...      ...    ...         ...    ...  \n","425    CB          NaN      NaN    NaN         NaN      1  \n","426   CD2          NaN      NaN    NaN         NaN      1  \n","427   OD2  C:746:_:ARG      NaN    NaN         NaN      1  \n","428    NZ  C:747:_:TYR      NaN    NaN         NaN      1  \n","429    CG          NaN      NaN    NaN         NaN      1  \n","\n","[430 rows x 13 columns]"]},"execution_count":9,"metadata":{},"output_type":"execute_result"}],"source":["import pandas as pd\n","import networkx as nx\n","import matplotlib.pyplot as plt\n","\n","# Specify the path to your tab-separated file\n","tsv_file = 'results2/1yok.cif_ringEdges'\n","\n","# Read the tab-separated file into a DataFrame\n","df = pd.read_csv(tsv_file, sep='\\t')\n","df"]},{"cell_type":"markdown","id":"00e3a730-e9bb-45d7-bf00-3f33d8edef70","metadata":{"id":"00e3a730-e9bb-45d7-bf00-3f33d8edef70"},"source":["# The Rendered PSN of 1YOK using Networkx and Plotly:\n","---"]},{"cell_type":"code","execution_count":null,"id":"49963974-aefe-4b30-92e8-5e8dc67af2d4","metadata":{"id":"49963974-aefe-4b30-92e8-5e8dc67af2d4"},"outputs":[],"source":["import pandas as pd\n","import networkx as nx\n","import plotly.graph_objects as go\n","import warnings\n","import plotly.io as pio\n","\n","warnings.simplefilter(action='ignore', category=FutureWarning)\n","\n","# Specify the path to your tab-separated file\n","tsv_file = 'results2/1yok.cif_ringEdges'\n","\n","# Read the tab-separated file into a DataFrame\n","df = pd.read_csv(tsv_file, sep='\\t')\n","\n","# Create the graph from the DataFrame\n","G = nx.from_pandas_edgelist(df, 'NodeId1', 'NodeId2')\n","\n","# Create a force-directed layout using NetworkX's spring_layout\n","pos = nx.spring_layout(G, k=0.75, seed=42)  # Use a fixed seed for reproducibility\n","\n","# Extract the positions\n","x_nodes = [pos[node][0] for node in G.nodes()]\n","y_nodes = [pos[node][1] for node in G.nodes()]\n","\n","# Extract the edges\n","edge_trace = []\n","for edge in G.edges():\n","    x0, y0 = pos[edge[0]]\n","    x1, y1 = pos[edge[1]]\n","    edge_trace.append(go.Scatter(\n","        x=[x0, x1, None],\n","        y=[y0, y1, None],\n","        mode='lines',\n","        line=dict(width=1, color='gray'),\n","        hoverinfo='none'\n","    ))\n","\n","# Create node trace\n","node_trace = go.Scatter(\n","    x=x_nodes,\n","    y=y_nodes,\n","    mode='markers+text',\n","    text=[str(node) for node in G.nodes()],\n","    textposition=\"bottom center\",\n","    hoverinfo='text',\n","    marker=dict(\n","        showscale=False,\n","        color='orange',\n","        size=10,\n","        line=dict(width=1, color='black')\n","    )\n",")\n","\n","# Create the figure\n","fig = go.Figure(data=edge_trace + [node_trace],\n","                layout=go.Layout(\n","                    title='Protein Structure Network of 1YOK Using RING Server',\n","                    titlefont_size=16,\n","                    showlegend=False,\n","                    hovermode='closest',\n","                    margin=dict(b=20, l=5, r=5, t=40),\n","                    # xaxis=dict(showgrid=True, zeroline=False, showticklabels=False, gridcolor='blue'),\n","                    # yaxis=dict(showgrid=True, zeroline=False, showticklabels=False, gridcolor='blue'),\n","                    dragmode='select'\n","                ))\n","\n","# Update the layout to set width and height, and background color\n","fig.update_layout(\n","    width=1000,\n","    height=1000,\n","    plot_bgcolor='white',  # Set the background color to white\n","    paper_bgcolor='white',  # Set the color of the plotting area to white\n",")\n","\n","pio.write_html(fig, file='network_graph.html', auto_open=True)\n","# Show the figure\n","fig.show()\n"]},{"cell_type":"code","execution_count":null,"id":"4fdaac90-5a5d-41f9-ad40-fcd3673d8834","metadata":{"id":"4fdaac90-5a5d-41f9-ad40-fcd3673d8834","outputId":"0fac5971-fc91-4226-d011-41f8bf4e6ca6"},"outputs":[{"data":{"text/html":["\n","        <iframe\n","            width=\"100%\"\n","            height=\"1000\"\n","            src=\"network_graph.html\"\n","            frameborder=\"0\"\n","            allowfullscreen\n","            \n","        ></iframe>\n","        "],"text/plain":["<IPython.lib.display.IFrame at 0x7ff7a0813b50>"]},"execution_count":20,"metadata":{},"output_type":"execute_result"}],"source":["from IPython.display import IFrame\n","\n","# Embed the local HTML file inside an iframe\n","IFrame(src='network_graph.html', width='100%', height=1000)"]},{"cell_type":"code","execution_count":null,"id":"4f0003c4-7ef8-4367-a43c-dd8a85883edc","metadata":{"id":"4f0003c4-7ef8-4367-a43c-dd8a85883edc"},"outputs":[],"source":[]}],"metadata":{"kernelspec":{"display_name":"Python 3 (ipykernel)","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.10.14"},"colab":{"provenance":[{"file_id":"https://github.com/LastCodeBender42/Drug-Screening-Project/blob/main/03_pdb_to_psn_using_ring.ipynb","timestamp":1723331324729}]}},"nbformat":4,"nbformat_minor":5}